components/RuiBreadcrumbs/src/RuiBreadcrumbs.ts

```diff
@@ -8,111 +8,283 @@
 import {LitElement, html, property, CSSResultArray, TemplateResult} from 'lit-element';
 import {variables, layout} from './RuiBreadcrumbs.css'
 
-
+/**
+ * RuiBreadcrumbs provide a way for users to see where they are in the 
+ * websites IA, as well as providing an easy way to navigate to previously 
+ * visited areas
+ */
 export class RuiBreadcrumbs extends LitElement {
-	public constructor() {
-		super();
-		/**
-		 * set child array
-		 * */
-		this._childArray = [...this.children];
-	}
 
 	/**
-	 * Private properties to set number of crumbs before and after collapse element
-	 * */
-	private _childArray: any[] = [];
-	private _collapsedArray: any[] = [];
-	private _isExpanded: boolean = false;
-
-	public set expand(val) {
-		const oldVal = this._isExpanded;
-		this._isExpanded = val;
-		this.requestUpdate('expand', oldVal);
-	}
-
-	public get expand() { return this._isExpanded; }
+	 * A reference to the slotted seperator component (if provided)
+	 */
+	private _seperatorEl: HTMLSlotElement | null = null;
 
 	/**
-	 * The array of breadcrumbs
+	 * An array of the custom slotted crumb elements
 	 */
-	@property({type : String})
-	public crumbs?: string = '';
+	private _crumbs: HTMLElement[] = [];
 
 	/**
-	 * Maximum length of crumbs to show
+	 * The maximum number of breadcrumbs to display 
+	 * before truncating
 	 */
-	@property({type : String})
-	public maxCrumbs?: string = '';
+	private _maxCrumbs: number = 7;
```

Comment:
Why do we have a default value?

I understand from an IA perspective you shouldn't really want this. But I don't think we should specify a default max as it would make configuration problematic when a user does want to show all as they always need to calculate what this is before setting it

Reply:
@ricominten yeah good point, I'll set it to the number of given crumbs if not set so that all crumbs show by default